PayItForward.ConsoleClient.UnitTests/Web/Controllers/StoriesController/Donate_Should.cs

```diff
@@ -0,0 +1,251 @@
+        {
+            // Arrange
+
+            // Act
+            var result = this.storiesController.Donate(id: Guid.Empty);
+
+            // Assert
+            var redirectToActionResult = Assert.IsType<RedirectToActionResult>(result);
+            Assert.Equal("Stories", redirectToActionResult.ControllerName);
+            Assert.Equal("Details", redirectToActionResult.ActionName);
+        }
+
+        [Fact]
+        public void ReturnContent_WithStoryNotFound_WhenStoryNotFound()
+        {
+            // Arrange
+            this.storiesService.Setup(s => s.GetStoryById(this.storyId))
+                .Returns((StoryDTO)null);
+
+            // Act
+            var result = this.storiesController.Donate(this.storyId);
+
+            // Assert
+            var contentResult = Assert.IsType<ContentResult>(result);
+            Assert.Equal("Story not found.", contentResult.Content);
+        }
+
+        [Fact]
+
+        public void ReturnResultContent_WithNoUserToDonate_WhenDonatorIdIsNullOrEmpty()
+        {
+            // Arrange
+            this.storiesService.Setup(s => s.GetStoryById(this.storyId))
+                .Returns(this.HelperStoryDto().FirstOrDefault());
+            this.storiesController.ControllerContext = new ControllerContext
+            {
+                HttpContext = new DefaultHttpContext
+                {
+                    User = new ClaimsPrincipal(
+                    new ClaimsIdentity(
+                        new Claim[]
+                        {
+                            new Claim(ClaimTypes.NameIdentifier, "username")
+                        },
+                        "someAuthTypeName"))
+                }
+            };
+            this.usersService.Setup(u => u.GetUserById(" ")).Returns((UserDTO)null);
+
+            // Act
+            var result = this.storiesController.Donate(this.HelperDonateViewModel(), this.storyId);
+
+            // Assert
+            var contentResult = Assert.IsType<ContentResult>(result);
+            Assert.Equal("No user to donate.", contentResult.Content);
+        }
+
+        [Fact]
+        public void ReturnContentResult_WithStoryNotFound_WhenGivenInvalidModel()
+        {
+            // Arrange & Act
+            var result = this.storiesController.Donate(model: null, id: this.storyId);
+
+            // Assert
+            var contentResult = Assert.IsType<ContentResult>(result);
+            Assert.Equal("Story not found.", contentResult.Content);
+        }
+
+        [Fact]
+        public void ReturnContentResult_WithStoryNotFound_WhenIdIsNull()
+        {
+            // Arrange & Act
+            var result = this.storiesController.Donate(model: new DonateViewModel(), id: Guid.Empty);
+
+            // Assert
+            var contentResult = Assert.IsType<ContentResult>(result);
+            Assert.Equal("Story not found.", contentResult.Content);
+        }
+
+        [Fact]
+        public void CallGetUserByIdOnce()
```

Comment:
You dont need to check if these method are called as they return something that we should test. So there is not need of this test